packages/rpc-core/src/response-patcher-allowed-numeric-values.ts

```diff
@@ -67,6 +67,7 @@ export function getAllowedNumericKeypaths(): AllowedNumericKeypaths {
         ];
         memoizedKeypaths = {
             getAccountInfo: jsonParsedAccountsConfigs.map(c => ['value', ...c]),
+            getBlock: [['blockTime']],
```

Comment:
Given this can return transactions as JSON we'll need to include `jsonParsedAccountsConfigs` here too. Possibly `['transactions', KEYPATH_WILDCARD, ...c]`?

Reply:
it doesn't look like it does accounts as `jsonParsed`, just the transactions. I'm seeing this as a returned payload for `jsonParsed`:

```json
{
                "meta": {
                    "err": null,
                    "fee": 5000,
                    "innerInstructions": [],
                    "logMessages": [
                        "Program FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH invoke [1]",
                        "Program FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH consumed 582 of 200000 compute units",
                        "Program FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH success"
                    ],
                    "postBalances": [
                        452462555000,
                        23942400,
                        1169280,
                        1141440
                    ],
                    "postTokenBalances": [],
                    "preBalances": [
                        452462560000,
                        23942400,
                        1169280,
                        1141440
                    ],
                    "preTokenBalances": [],
                    "rewards": null,
                    "status": {
                        "Ok": null
                    }
                },
                "transaction": {
                    "message": {
                        "accountKeys": [
                            {
                                "pubkey": "JTmFx5zX9mM94itfk2nQcJnQQDPjcv4UPD7SYj6xDCV",
                                "signer": true,
                                "source": "transaction",
                                "writable": true
                            },
                            {
                                "pubkey": "B47CC1ULLw1jKTSsr1N1198zrUHp3LPduzepJyzgLn2g",
                                "signer": false,
                                "source": "transaction",
                                "writable": true
                            },
                            {
                                "pubkey": "SysvarC1ock11111111111111111111111111111111",
                                "signer": false,
                                "source": "transaction",
                                "writable": false
                            },
                            {
                                "pubkey": "FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH",
                                "signer": false,
                                "source": "transaction",
                                "writable": false
                            }
                        ],
                        "instructions": [
                            {
                                "accounts": [
                                    "JTmFx5zX9mM94itfk2nQcJnQQDPjcv4UPD7SYj6xDCV",
                                    "B47CC1ULLw1jKTSsr1N1198zrUHp3LPduzepJyzgLn2g",
                                    "SysvarC1ock11111111111111111111111111111111"
                                ],
                                "data": "6mJFQAEssWECZ33ewGbTZCAexvd6ohfcq3DK8V1htvVooao8Jdqwhy",
                                "programId": "FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH",
                                "stackHeight": null
                            }
                        ],
                        "recentBlockhash": "9eP43idbS58zQPhQvmNAHaLJcu9qVxbfrYMSMKk6AcMP"
                    },
                    "signatures": [
                        "3h1Sby9n6U9UMKempuuAwa4VSFnELgddXcNHKe2379yBaYpDYKUrrqAUTah2B9vdZvVCE1fE4bPu3AHvCrBP8V3t"
                    ]
                },
                "version": "legacy"
            },
```

Reply:
I think you're right, got confused - `getTransaction` doesn't pull in all the account stuff either. 

But I think at least some of the `getTransaction` overrides are relevant

If you do this call on devnet:

```json
{
    "method": "getBlock",
    "jsonrpc": "2.0",
    "params": [
        243738851,
        {
            "commitment": "confirmed",
            "maxSupportedTransactionVersion": 0,
            "transactionDetails": "full",
            "encoding": "jsonParsed",
            "rewards": true
        }
    ],
    "id": "0e5f2a86-413e-4d03-b9ba-17b4b5cb4d3e"
}
```

Then the first transaction (signature: 62v4UFxaAtyLu35bQdHb8WGZwJxpzbYvaNx2GPvrkag8PWvWRV5v8ADTA4gEvHzeWtf67zuufqVTtSiSML6dCXJP) has this meta:

```json
                "meta": {
                    "computeUnitsConsumed": 135487,
                    "err": null,
                    "fee": 5003,
                    "innerInstructions": [
                        {
                            "index": 2,
                            "instructions": [
                                {
                                    "parsed": {
                                        "info": {
                                            "amount": "10",
                                            "authority": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                                            "destination": "CYfbTCjJVpLjvZPcoh63YMugrKchYmRQKseQNS2UK2or",
                                            "source": "8NqJAQxy8eyAX3k2egLFwaVegn9cf9U9xBKZSq94Th5o"
                                        },
                                        "type": "transfer"
                                    },
                                    "program": "spl-token",
                                    "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                                    "stackHeight": 2
                                },
                                {
                                    "parsed": {
                                        "info": {
                                            "amount": "12500",
                                            "authority": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                                            "destination": "CYfbTCjJVpLjvZPcoh63YMugrKchYmRQKseQNS2UK2or",
                                            "source": "8NqJAQxy8eyAX3k2egLFwaVegn9cf9U9xBKZSq94Th5o"
                                        },
                                        "type": "transfer"
                                    },
                                    "program": "spl-token",
                                    "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                                    "stackHeight": 2
                                }
                            ]
                        }
                    ],
                    "logMessages": [
                        "(snipped for brevity)"
                    ],
                    "postBalances": [
                        1519905907862,
                        ...
                    ],
                    "postTokenBalances": [
                        {
                            "accountIndex": 6,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40618520620",
                                "decimals": 9,
                                "uiAmount": 40.61852062,
                                "uiAmountString": "40.61852062"
                            }
                        },
                        {
                            "accountIndex": 8,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "8168126929",
                                "decimals": 9,
                                "uiAmount": 8.168126929,
                                "uiAmountString": "8.168126929"
                            }
                        },
                        {
                            "accountIndex": 9,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40704553815",
                                "decimals": 9,
                                "uiAmount": 40.704553815,
                                "uiAmountString": "40.704553815"
                            }
                        },
                        {
                            "accountIndex": 10,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40674737185",
                                "decimals": 9,
                                "uiAmount": 40.674737185,
                                "uiAmountString": "40.674737185"
                            }
                        }
                    ],
                    "preBalances": [
                        1519905912865,
                        ...
                    ],
                    "preTokenBalances": [
                        {
                            "accountIndex": 6,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40618520620",
                                "decimals": 9,
                                "uiAmount": 40.61852062,
                                "uiAmountString": "40.61852062"
                            }
                        },
                        {
                            "accountIndex": 8,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "8168139439",
                                "decimals": 9,
                                "uiAmount": 8.168139439,
                                "uiAmountString": "8.168139439"
                            }
                        },
                        {
                            "accountIndex": 9,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40704541305",
                                "decimals": 9,
                                "uiAmount": 40.704541305,
                                "uiAmountString": "40.704541305"
                            }
                        },
                        {
                            "accountIndex": 10,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40674737185",
                                "decimals": 9,
                                "uiAmount": 40.674737185,
                                "uiAmountString": "40.674737185"
                            }
                        }
                    ],
                    "rewards": [],
                    "status": {
                        "Ok": null
                    }
                },
```

So the bits in innerInstructions and pre/post token balances can be returned here when using `jsonParsed`

I think the stuff there under `innerInstructions.parsed` is missing for `getTransaction` too

Also if you request the same block with `json` encoding, you'll see that we also need most of the other overrides under `getTransaction`:

```json
            {
                "meta": {
                    "computeUnitsConsumed": 135487,
                    "err": null,
                    "fee": 5003,
                    "innerInstructions": [
                        {
                            "index": 2,
                            "instructions": [
                                {
                                    "accounts": [
                                        8,
                                        9,
                                        15
                                    ],
                                    "data": "3F8aKq24k59R",
                                    "programIdIndex": 19,
                                    "stackHeight": 2
                                },
                                {
                                    "accounts": [
                                        8,
                                        9,
                                        15
                                    ],
                                    "data": "3px3hhrW2tYw",
                                    "programIdIndex": 19,
                                    "stackHeight": 2
                                }
                            ]
                        }
                    ],
                    "loadedAddresses": {
                        "readonly": [],
                        "writable": []
                    },
                    "logMessages": [
                        "<snip>"
                    ],
                    "postBalances": [
                        1519905907862,
                        ...
                    ],
                    "postTokenBalances": [
                        {
                            "accountIndex": 6,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40618520620",
                                "decimals": 9,
                                "uiAmount": 40.61852062,
                                "uiAmountString": "40.61852062"
                            }
                        },
                        {
                            "accountIndex": 8,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "8168126929",
                                "decimals": 9,
                                "uiAmount": 8.168126929,
                                "uiAmountString": "8.168126929"
                            }
                        },
                        {
                            "accountIndex": 9,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40704553815",
                                "decimals": 9,
                                "uiAmount": 40.704553815,
                                "uiAmountString": "40.704553815"
                            }
                        },
                        {
                            "accountIndex": 10,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40674737185",
                                "decimals": 9,
                                "uiAmount": 40.674737185,
                                "uiAmountString": "40.674737185"
                            }
                        }
                    ],
                    "preBalances": [
                        1519905912865,
                        ...
                    ],
                    "preTokenBalances": [
                        {
                            "accountIndex": 6,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40618520620",
                                "decimals": 9,
                                "uiAmount": 40.61852062,
                                "uiAmountString": "40.61852062"
                            }
                        },
                        {
                            "accountIndex": 8,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "8168139439",
                                "decimals": 9,
                                "uiAmount": 8.168139439,
                                "uiAmountString": "8.168139439"
                            }
                        },
                        {
                            "accountIndex": 9,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40704541305",
                                "decimals": 9,
                                "uiAmount": 40.704541305,
                                "uiAmountString": "40.704541305"
                            }
                        },
                        {
                            "accountIndex": 10,
                            "mint": "So11111111111111111111111111111111111111112",
                            "owner": "CyZuD7RPDcrqCGbNvLCyqk6Py9cEZTKmNKujfPi3ynDd",
                            "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
                            "uiTokenAmount": {
                                "amount": "40674737185",
                                "decimals": 9,
                                "uiAmount": 40.674737185,
                                "uiAmountString": "40.674737185"
                            }
                        }
                    ],
                    "rewards": [],
                    "status": {
                        "Ok": null
                    }
                },
                "transaction": {
                    "message": {
                        "accountKeys": [
                            "2KgowxogBrGqRcgXQEmqFvC3PGtCu66qERNJevYW8Ajh",
                            ...
                        ],
                        "header": {
                            "numReadonlySignedAccounts": 0,
                            "numReadonlyUnsignedAccounts": 7,
                            "numRequiredSignatures": 1
                        },
                        "instructions": [
                            {
                                "accounts": [],
                                "data": "3FTyfrdhjHgT",
                                "programIdIndex": 14,
                                "stackHeight": null
                            },
                            {
                                "accounts": [],
                                "data": "HnkkG7",
                                "programIdIndex": 14,
                                "stackHeight": null
                            },
                            {
                                "accounts": [
                                    1,
                                    3,
                                    0,
                                    16,
                                    0,
                                    5,
                                    13,
                                    12,
                                    8,
                                    19,
                                    15,
                                    2,
                                    17,
                                    7,
                                    4,
                                    3,
                                    10,
                                    6,
                                    9,
                                    11
                                ],
                                "data": "3hAw8ppjnbTHztHcRqreNUA4g9AGRBcya8ydZixoM59HsjYE2szjsytfdyZrFL3uPNXMFdJjLAfQg2oBTNgnKJY6urM5Jq1NktctfKQTVH4xNLHfPfRr89gvdD1Pds73W2652Ws5DYA8kWagWRmYL",
                                "programIdIndex": 18,
                                "stackHeight": null
                            }
                        ],
                        "recentBlockhash": "6Kr6yCUeXihUSH5CDMP3zeuryiiVm24Jbcna1vs8aQCZ"
                    },
                    "signatures": [
                        "62v4UFxaAtyLu35bQdHb8WGZwJxpzbYvaNx2GPvrkag8PWvWRV5v8ADTA4gEvHzeWtf67zuufqVTtSiSML6dCXJP"
                    ]
                },
                "version": "legacy"
            },
```

That covers a bunch of other `getTransaction` overrides that we'll need here. We probably need a variant of all of them, not the account ones, sorry for the mixup!